app/src/main/java/org/oppia/android/app/databinding/MarginBindingAdapters.java

```diff
@@ -1,35 +1,61 @@
 package org.oppia.android.app.databinding;
 
+import android.annotation.SuppressLint;
 import android.view.View;
 import android.view.ViewGroup.MarginLayoutParams;
+
 import androidx.annotation.NonNull;
+import androidx.core.view.ViewCompat;
 import androidx.databinding.BindingAdapter;
 
-/** Holds all custom binding adapters that set margin values. */
+/**
+ * Holds all custom binding adapters that set margin values.
+ */
 public final class MarginBindingAdapters {
 
-  /** Used to set a margin-start for views. */
+  @SuppressLint("WrongConstant")
+  private static boolean isRtlLayout(View view) {
+    return view.getLayoutDirection() == ViewCompat.LAYOUT_DIRECTION_RTL;
+  }
+
+  /**
+   * Used to set a margin-start for views.
+   */
   @BindingAdapter("app:layoutMarginStart")
   public static void setLayoutMarginStart(@NonNull View view, float marginStart) {
     if (view.getLayoutParams() instanceof MarginLayoutParams) {
       MarginLayoutParams params = (MarginLayoutParams) view.getLayoutParams();
+      float marginEnd = params.getMarginEnd();
+      if (isRtlLayout(view)) {
+        float temp = marginStart;
+        marginStart = marginEnd;
+        marginEnd = temp;
+      }
       params.setMargins(
           (int) marginStart,
           params.topMargin,
-          params.getMarginEnd(),
+          (int) marginEnd,
           params.bottomMargin
       );
```

Comment:
Maybe consider instead creating a helper method to set margins which automatically takes into consideration RTL. That will avoid needing an if-check in each binding method since they'd share a common margin value setter.

Reply:
Done.